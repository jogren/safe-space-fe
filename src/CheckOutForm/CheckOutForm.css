.CheckOutForm {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 50%;
  margin: auto;
  height: 85vh;
  border: 2px solid black;
  background-color: black;
  opacity: .7;
  margin-top: 40px;
  color: white;
  border-radius: 10px;
  font-weight: bold;
  overflow-y: scroll;
}

.CheckOutForm_input {
  height: 50px;
  border-radius: 5px;
  margin-top: 20px;
  width: 50%;
  font-size: 20px;
  padding-left: 5px;
}

.CheckOutForm_visitor {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
  font-family: 'Aclonica';
  font-weight: bold;
  border-color: white; 
}

#delete-visitor {
  cursor: pointer; 
}

.CheckOutForm_visitor-container {
  margin-top: 20px;
  box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
  border: 1px solid white;
  padding: 20px;
  width: 90%;
}

/* 
.button {
  color: white;
  text-align: center;
  font-size: 16px;
  opacity: 0.6;
  transition: 0.3s;
  text-decoration: none;
}

.button:hover {opacity: 4}

.push {
  margin:0.4em; 
}

#push:hover{
  animation: pulse 0.7s linear 1;
} */