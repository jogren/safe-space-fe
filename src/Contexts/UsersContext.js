import { createContext } from 'react';

export const UsersContext = createContext([]);

export const Provider = props => {
  
}