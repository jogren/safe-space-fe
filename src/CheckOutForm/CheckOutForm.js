import React, { useState, useContext } from "react";
import { UsersContext } from "../Contexts/UsersContext";
import { deleteUserFromDB } from "../apiCalls/apiCalls";
import "../CheckOutForm/CheckOutForm.css";

const CheckOutForm = () => {
  const { currentUsers, setCurrentUsers } = useContext(UsersContext);
  const [search, updateSearch] = useState("");

  const deleteUser = async id => {
    await deleteUserFromDB(id);
    let filteredUsers = currentUsers.original.filter(person => person.id != id);
    setCurrentUsers({ result: filteredUsers, original: filteredUsers });
  };

  let namesArray = !search ? currentUsers.original : currentUsers.result;
  let namesList = namesArray.map((name, index) => {
    return (
      <div className="CheckOutForm_visitor button push" id="push" key={index}>
        <p id="person-name">{name.name}</p>
        <button id="delete-visitor-button" onClick={() => deleteUser(name.id)}>
          <img
            id="delete-visitor-button-img"
            alt="circle with an x inside"
            src={require("../assets/delete-button.svg")}
          />
        </button>
      </div>
    );
  });

  const handleSearch = e => {
    updateSearch(e.target.value);
    if (!e.target.value) {
      setCurrentUsers({
        original: currentUsers.original,
        result: currentUsers.original
      });
    } else {
      let filteredNames = currentUsers.result.filter(({ name }) =>
        name.toLowerCase().includes(e.target.value.toLowerCase())
      );
      setCurrentUsers({ ...currentUsers, result: filteredNames });
    }
  };

  return (
    <section className="CheckOutForm">
      <h1 id="checkout-form-text">Check-Out Visitor</h1>
      <input
        className="CheckOutForm_input"
        type="text"
        placeholder="Search the database..."
        name="search"
        value={search}
        onChange={handleSearch}
        autoComplete="off"
      />
      <div className="CheckOutForm_visitor-container">{namesList}</div>
    </section>
  );
};

export default CheckOutForm;
