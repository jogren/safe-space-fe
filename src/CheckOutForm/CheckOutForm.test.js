import React from "react";
import { mount } from "enzyme";
import CheckOutForm from "./CheckOutForm";
import { UsersContext } from '../Contexts/UsersContext';
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("CheckOutForm", () => {
  let wrapper;
  let UsersContext;

  beforeEach(() => {
    UsersContext = {
      reliefCenter: {
        name: "Turing Relief Center",
        addressPrint: "1331 17th St Denver CO 80202",
        lat: "39.7508132",
        lng: "-104.9967997",
        phone: "5555555555",
        website: "www.relief-center.com",
        id: "1"
      },
      setreliefCenter: () => {}            
    }
    wrapper = mount(
      <CheckOutForm/>,
      { UsersContext }
    );
  });

  it("should render without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should mock context", () => {
    wrapper.setContext({UsersContext})
    expect(wrapper).toMatchSnapshot();
  });
});