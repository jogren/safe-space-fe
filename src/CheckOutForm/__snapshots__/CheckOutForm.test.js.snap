// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOutForm should match the snapshot 1`] = `undefined`;

exports[`CheckOutForm should match the snapshot 2`] = `undefined`;

exports[`CheckOutForm should mock context 1`] = `undefined`;
